src/assets/scripts/client/trafficGenerator/buildPreSpawnAircraft.js

```diff
@@ -1,78 +1,245 @@
+            // }
+
+        const nextWaypointIndex = _findIndex(waypointOffsetMap, (distanceToWaypoint) => {
+            return distanceToWaypoint >= spawnOffset;
+        });
+        const nextWaypointModel = waypointModelList[nextWaypointIndex];
+        const previousWaypointIndex = Math.max(0, nextWaypointIndex - 1);
+        const previousWaypointModel = waypointModelList[previousWaypointIndex];
+        const distanceFromPreviousWaypointToSpawnPoint = spawnOffset - waypointOffsetMap[previousWaypointIndex];
+        const heading = previousWaypointModel.calculateBearingToWaypoint(nextWaypointModel);
+        const spawnPositionModel = previousWaypointModel.positionModel.generateDynamicPositionFromBearingAndDistance(
+            heading,
+            distanceFromPreviousWaypointToSpawnPoint
+        );
+        const altitude = _calculateAltitudeAtOffset(waypointModelList, waypointOffsetMap, altitudeOffsets, spawnOffset);
+
+        spawnPositionsAndAltitudes.push({
+            altitude,
+            heading,
+            nextFix: nextWaypointModel.name,
+            positionModel: spawnPositionModel
+        });
     }
 
-    return spawnPositions;
-};
+    return spawnPositionsAndAltitudes;
+}
 
 /**
- * Calculate distance(s) from center where an aircraft should exist onload or airport change
+ * Calculate distances along spawn pattern route at which to prespawn aircraft
+ *
+ * To randomize the spawn locations, the interval between aircraft will vary, but should
+ * average out to exactly the `entrailDistance`. The exception is if the `entrailDistance`
+ * is less than the `smallestIntervalNm` defined below. In that case, aircraft will be
+ * spawned at exactly the `entrailDistance` with no variation due to their proximity.
+ * 
+ * NOTE: Provided there is at least `smallestIntervalNm` distance between them, an aircraft
+ * will always be spawned right along the airspace boundary, and another at the first fix.
+ *
+ * For example, with `smallestIntervalNm = 15`:
+ *   - If requesting 8MIT, will spawn exactly 8MIT
+ *   - If requesting 30MIT, will spawn each a/c 15MIT-45MIT of the previous arrival
  *
  * @function _assembleSpawnOffsets
  * @param entrailDistance {number}
  * @param totalDistance {number}
- * @return spawnOffsets {array<number>}
+ * @return spawnOffsets {array<number>} distances along route, in nm
  */
 const _assembleSpawnOffsets = (entrailDistance, totalDistance = 0) => {
-    const spawnOffsets = [];
+    const offsetClosestToAirspace = totalDistance - 3;
+    let smallestIntervalNm = 15;
+    const largestIntervalNm = entrailDistance + (entrailDistance - smallestIntervalNm);
+
+    // if requesting less than `smallestIntervalNm`, spawn all AT `entrailDistance`
+    if (smallestIntervalNm > largestIntervalNm) {
+        smallestIntervalNm = largestIntervalNm;
+    }
+
+    const spawnOffsets = [offsetClosestToAirspace];
+    let offset = offsetClosestToAirspace;
```

Comment:
Is there a better name here than just offset

Reply:
`distanceAlongRoute`. Not great, but better I think